contracts/BullaFinance.sol

```diff
@@ -39,18 +62,26 @@ abstract contract BullaFinance {
     /// @notice SPEC:
     ///     allows an admin to change the admin address to `_admin`
     ///     Given the following: `msg.sender == admin`
-    function changeAdmin(address _admin) public virtual;
+    // function changeAdmin(address _admin) public virtual;
 
     /// @param _fee the new fee
     /// @notice SPEC:
     ///     allows an admin to update the fee amount to `_fee`
     ///     Given the following: `msg.sender == admin`
-    function feeChanged(uint256 _fee) public virtual;
+    // function feeChanged(uint256 _fee) public virtual;
 
     /// @notice SPEC:
     ///     allows an admin to withdraw `withdrawableFee` amount of tokens from this contract's balance
     ///     Given the following: `msg.sender == admin`
-    function withdrawFee() public virtual;
+    function withdrawFee() public {
+        if (msg.sender != admin) revert NOT_ADMIN();
+
+        uint256 withdrawableFee = fee * unharvestedTransactions;
```

Comment:
don't want to give a brain stroke, but what happens when we change fee and fees were already deposited?

Reply:
Wow great catch. Yeah it would be insolvent if the fee increases, or fee would be stuck. 
Perhaps the best move here is to just pass and amount in and trust the admin anyways

Reply:
maybe keep a mapping of unharvestedTransactionCountByFeeAmount